src/main/java/org/trigon/hardware/misc/leds/CANdleLEDStrip.java

```diff
@@ -0,0 +1,212 @@
+        if (RobotHardwareStats.isSimulation()) {
+            simulationLEDStrip.blink(firstColor, secondColor, blinkingIntervalSeconds);
+            return;
+        }
+        CANDLE.animate(
+                new SingleFadeAnimation(
+                        (int) firstColor.red,
+                        (int) firstColor.green,
+                        (int) firstColor.blue,
+                        0,
+                        blinkingIntervalSeconds,
+                        this.numberOfLEDs,
+                        indexOffset
+                ),
+                animationSlot
+        );
+    }
+
+    @Override
+    void staticColor(Color color) {
+        if (RobotHardwareStats.isSimulation()) {
+            simulationLEDStrip.staticColor(color);
+            return;
+        }
+        CANDLE.setLEDs((int) color.red, (int) color.green, (int) color.blue, 0, indexOffset, numberOfLEDs);
+    }
+
+    @Override
+    void breathe(Color color, int breathingLEDs, double cycleTimeSeconds, boolean shouldLoop, boolean inverted, LarsonAnimation.BounceMode bounceMode) {
+        if (RobotHardwareStats.isSimulation()) {
+            simulationLEDStrip.breathe(color, breathingLEDs, cycleTimeSeconds, shouldLoop, inverted, bounceMode);
+            return;
+        }
+        CANDLE.animate(
+                new LarsonAnimation(
+                        (int) color.red,
+                        (int) color.green,
+                        (int) color.blue,
+                        0,
+                        cycleTimeSeconds,
+                        this.numberOfLEDs,
+                        bounceMode,
+                        breathingLEDs,
+                        indexOffset
+                ),
+                animationSlot
+        );
+    }
+
+    @Override
+    void alternateColor(Color firstColor, Color secondColor, double intervalSeconds) {
+        if (RobotHardwareStats.isSimulation()) {
+            simulationLEDStrip.alternateColor(firstColor, secondColor, intervalSeconds);
+            return;
+        }
+        if (Timer.getFPGATimestamp() - lastAlternateColorTime > intervalSeconds) {
+            alternateColor = !alternateColor;
+            lastAlternateColorTime = Timer.getFPGATimestamp();
+        }
+        if (alternateColor) {
+            for (int i = 0; i < numberOfLEDs; i++)
+                CANDLE.setLEDs(
+                        (int) firstColor.red,
+                        (int) firstColor.green,
+                        (int) firstColor.blue,
+                        0,
+                        i + indexOffset,
+                        1
+                );
+            return;
+        }
+        for (int i = 0; i < numberOfLEDs; i++)
+            CANDLE.setLEDs(
+                    (int) firstColor.red,
+                    (int) firstColor.green,
+                    (int) firstColor.blue,
+                    0,
+                    i + indexOffset,
+                    1
+            );
+    }
```

Comment:
Double check all of this. Also, isn't this just the blink command?

Reply:
I checked it, and it's not the blink command